nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -640,14 +635,38 @@ in
       preStart =
         ''
         ${cfg.preStart}
-        ${cfg.package}/bin/nginx -c ${configFile} -p ${cfg.stateDir} -t
+        ${cfg.package}/bin/nginx -c '${configFile}' -p '/var/lib/${cfg.stateDirName}' -t
         '';
       serviceConfig = {
-        ExecStart = "${cfg.package}/bin/nginx -c ${configFile} -p ${cfg.stateDir}";
+        ExecStart = "${cfg.package}/bin/nginx -c '${configFile}' -p '/var/lib/${cfg.stateDirName}'";
         ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
         Restart = "always";
         RestartSec = "10s";
         StartLimitInterval = "1min";
+        # User and group
+        User = cfg.user;
+        Group = cfg.group;
+        # Filesystem access
+        ProtectSystem = "strict";
```

Comment:
If all the filesystem is read-only, how to handle caching? http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cache_path

Reply:
The state directory is writable. Isn't this exactly what the state directory is for (honest question, I never used the cache)?
Also, if you need a tmpfs filesystem (because performance > persistence), you get a (private) /tmp

Reply:
You're right, I was confused since there is no official concept of "state directory" (afaik) with Nginx (I don't know how they call this folder, or if it even has a name) but on Debian you would usually use `/var/lib/nginx` for this or `/tmp`, so it's ok. 👍